app/assets/js/rtm-upload-terms.js

```diff
@@ -1,139 +1,169 @@
+            rtp_display_terms_warning( terms_conditions_checkbox.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
+
+            return false;
+        } else {
+
+            return true;
+        }
+
+    } );
+
+    /**
+     * Show the warning message if the terms checkbox is unchecked before posting the media update.
+     */
+    rtMediaHook.register( 'rtmedia_js_before_activity_added', function ( args ) {
+        var terms_conditions_checkbox, form;
+        var whats_new_submit = jQuery( '#aw-whats-new-submit' );
+
+        if ( args && undefined !== args.src && 'activity' === args.src ) {
+            form = jQuery( '#whats-new-form' );
+            terms_conditions_checkbox = form.find( '#rtmedia_upload_terms_conditions' );
+        } else {
+            terms_conditions_checkbox = jQuery( '#rtmedia_upload_terms_conditions' );
+        }
+
+        if ( 1 === terms_conditions_checkbox.length ) {
+            terms_conditions_checkbox.removeAttr( 'disabled' );
+
+            if ( false === args ) {
+                whats_new_submit.removeAttr( 'disabled' );
+                whats_new_submit.removeClass( 'loading' );
+
+                return args;
+            }
+
+            if ( ! terms_conditions_checkbox.is( ':checked' ) ) {
+                whats_new_submit.removeAttr( 'disabled' );
+                whats_new_submit.removeClass( 'loading' );
+                if ( undefined !== args && false !== args && undefined !== args.src && 'activity' === args.src ) {
+                    rtp_display_terms_warning( form.find('.rtmedia-upload-terms'), rtmedia_upload_terms_data.message );
+                } else {
+                    rtp_display_terms_warning( terms_conditions_checkbox.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
+                }
+
+                return false;
+            }
+        }
+
+        return true;
+    } );
+
+    /**
+     * When Select Attribute for media [ rtmedia-custom-attributes: Add-Ons ] Issue:8,
+     * This Hook returns false that's why the attributes were not saved,
+     * Added: 'return true;'
+     * By: Yahil
+     */
+    rtMediaHook.register( 'rtmedia_js_after_file_upload', function () {
+
+        var terms_conditions_checkbox = jQuery( '#rtmedia-upload-container #rtmedia_upload_terms_conditions' );
+        if ( 1 === terms_conditions_checkbox.length ) {
+            terms_conditions_checkbox.removeAttr( 'checked' );
+            jQuery( '.rt_alert_msg' ).remove();
+        }
+
+        return true;
+    });
+
+    /**
+     * Uncheck the terms checkbox after the activity is posted successfully.
+     */
+    rtMediaHook.register( 'rtmedia_js_after_activity_added', function () {
+        var rtmedia_terms_conditions = $( '#rtmedia_upload_terms_conditions' );
+        if ( 1 === rtmedia_terms_conditions.length && rtmedia_terms_conditions.is( ':checked' ) ) {
```

Comment:
No need to check any condition, just uncheck the checkbox.
Refer - https://stackoverflow.com/questions/38751168/do-i-really-need-to-check-if-an-element-exists-with-jquery#answer-38751351